### PR TITLE
Import ABC from collections.abc for Python 3.10 compatibility.

### LINKED ISSUE #14013: Importing ABC directly from collections was deprecated and will be removed in Python 3.10
**Describe the current behavior**  

Importing ABC directly from collections has been deprecated since Python 3.4 and will be removed in Python 3.10. Use collections.abc in Python 3 and collections in Python 2.

```
keras/callbacks/callbacks.py
19:from collections import Iterable

keras/engine/training_utils.py
639:    if not isinstance(validation_freq, collections.Container):
641:                         '`collections.Container` (e.g. list, tuple, etc.)')
696:    if isinstance(loss, collections.Mapping):
811:    if isinstance(loss, collections.Mapping):
823:    elif isinstance(loss, collections.Sequence):
857:    elif isinstance(loss_weights, collections.Mapping):
927:    elif isinstance(metrics, collections.Mapping):

```

### PR DESCRIPTION
**Pull request by tirkarthi**
Fixes #14013

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thanks

**Comment by fchollet** [comment]
You'll need to rebase on master first, before we can merge.

**Comment by gbaned** [comment]
@tirkarthi  Can you please check @fchollet's comments and keep us posted ? Thanks!

**Comment by tirkarthi** [comment]
I will be away from computer for 2 weeks. I have updated branch through GitHub UI.

File: keras/feature_column/base_feature_layer.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 import tensorflow.compat.v2 as tf
 
-import collections
+import collections.abc
```
Comment (fchollet) [inline keras/feature_column/base_feature_layer.py:26]: Surely you should also be updating the locations that access `collections` members in this file?

**Review by fchollet — CHANGES_REQUESTED** [review]

File: keras/feature_column/base_feature_layer.py (hunk 1 above)
Comment (tirkarthi) [reply keras/feature_column/base_feature_layer.py:26]: It seems someone else has already updated the line in https://github.com/keras-team/keras/commit/5a15df56673aa5db7841c58a1565c0605f28c271 . I also made the change but I guess as part of the merge it got resolved by itself. I think this PR can be closed.

**Pull request closed without merging**